pkg/apis/core/validation/validation.go

```diff
@@ -3493,6 +3508,21 @@ func ValidatePodStatusUpdate(newPod, oldPod *core.Pod) field.ErrorList {
 	return allErrs
 }
 
+// validatePodConditions tests if the custom pod conditions are valid.
+func validatePodConditions(conditions []core.PodCondition, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	systemConditions := sets.NewString(string(core.PodScheduled), string(core.PodReady), string(core.PodInitialized))
+	for i, condition := range conditions {
+		if systemConditions.Has(string(condition.Type)) {
+			continue
+		}
+		for _, msg := range validation.IsQualifiedName(string(condition.Type)) {
```

Comment:
This doesn't actually check for a fully-qualified name - `foobar` is allowed as is `example.com/foobar`.   You maybe want  something different?

Reply:
also want a testcase, I think